src/engine/Input/Pointer.ts

```diff
@@ -321,106 +327,254 @@ export class Pointers extends Class {
    }
 
    /**
-    * Propogates events to actor if necessary
+    * Propogates events thorugh event path if necessary
     */
-   public propogate(actor: Actor) {
-      var isNotUIActor = !Actors.isUIActor(actor);
-      var i: number = 0;
-      var len: number = 0;
-      var pointerEvent: PointerEvent;
-      var pointer: Pointer;
+   public propagate() {
+      this._propagatePointerEvent('pointerdown', this._pointerDown);
+      this._propagatePointerEvent('pointerup', this._pointerUp);
+      this._propagatePointerEvent('pointermove', this._pointerMove);
+      this._propagatePointerEvent('pointercancel', this._pointerCancel);
+      this._propagateWheelEvent('pointerwheel', this._wheel);
+   }
+
+   /**
+    * Verifies Actor pointer events accordingly to actor provided
+    * @param actor  Actor to be verified
+    */
+   public verifyPointerEvents(actor: Actor) {
```

Comment:
`verify` implies that the method is not side-effecting. `propagate` may be a better verb here

Is there a difference between `propagate` and `verifyPointerEvents`? See refactor proposal below